src/v2/view-builder/views/DeviceChallengePollView.js

```diff
@@ -0,0 +1,155 @@
+
+const Body = BaseForm.extend({
+  noButtonBar: true,
+  initialize () {
+    this.deviceChallengePollRemediation = this.options.appState.get('currentState').remediation.find(
+      rem => rem.name === 'device-challenge-poll'
+    );
+    this.doChallenge();
+    BaseForm.prototype.initialize.apply(this, arguments);
+    this.startPolling();
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    this.stopPolling();
+  },
+
+  doChallenge () {
+    const deviceChallengeHintsKey = this.deviceChallengePollRemediation.relatesTo;
+    const deviceChallenge = deviceChallengeHintsKey &&
+      this.options.appState.get('currentState')[deviceChallengeHintsKey];
+    switch (deviceChallenge.challengeMethod) {
+    case 'loopback':
+      this.doLoopback(deviceChallenge.domain, deviceChallenge.ports, deviceChallenge.challengeRequest);
+      break;
+    case 'custom-scheme':
+      this.doCustomURI(deviceChallenge.href);
+      break;
+    }
+  },
+
+  doLoopback (authenticatorDomainUrl = '', ports = [], challengeRequest = '') {
+    let currentPort;
+    let foundPort = false;
+
+    const getAuthenticatorUrl = (path) => {
+      return `https://${authenticatorDomainUrl}:${currentPort}/${path}`;
+    };
+
+    const checkPort = () => {
+      return request(getAuthenticatorUrl('probe'));
+    };
+
+    const onPortFound = () => {
+      foundPort = true;
+      return request(getAuthenticatorUrl('challenge'), 'POST', JSON.stringify({ challengeRequest }) , 3000);
+    };
+
+    const onFailure = () => {};
+
+    const doProbing = () => {
+      return checkPort()
+        .then(onPortFound)
+        .catch(onFailure);
+    };
+
+    let probeChain = Promise.resolve();
+    ports.forEach(port => {
+      probeChain = probeChain
+        .then(() => {
+          // console.log(`foundPort: ${foundPort}`);
+          if (!foundPort) {
+            currentPort = port;
+            return doProbing();
+          }
+        })
+        .catch(() => {
+          // console.log('Something unexpected happened during device probing.', 'error');
+        });
+    });
+  },
+
+  doCustomURI (customURI) {
+    // this.add(`<a href="${customURI}" id="launch-app">custom URI here</a>`);
+    // this.$('#launch-app').click();
+  },
+
+  startPolling () {
+    const deviceChallengePollingInterval = this.deviceChallengePollRemediation &&
+      this.deviceChallengePollRemediation.refresh;
+    if (_.isNumber(deviceChallengePollingInterval)) {
+      // TODO: how to show spinner to indicating action in progress?
+      this.polling = setInterval(() => {
+        this.options.appState.trigger('saveForm', this.model);
+      }, deviceChallengePollingInterval);
+      // TODO: remove before release
+      setTimeout(_.bind(() => {this.stopPolling();}, this), 5000);
+    }
+  },
+
+  stopPolling () {
+    if (this.polling) {
+      console.log('stop polling');
+      clearInterval(this.polling);
+    }
+  },
+});
+
+const Footer = BaseFooter.extend({
```

Comment:
looks it's identical to Identify.footer?
we could do
```
import IdentifierView from './views/IdentifierView';
const Footer = Identifier.prototype.Footer;
```

Reply:
Dont worry this part yet. I just copied from some other view and forgot to remove it.